File: to-do-app/To-Do-Java/src/main/java/com/renan/todo/controller/TaskListController.java
package com.renan.todo.controller;

import com.renan.todo.controller.filter.AuthFilter;
import com.renan.todo.controller.form.TaskListForm;
import com.renan.todo.dto.ErrorDTO;
import com.renan.todo.dto.TaskListDTO;
import com.renan.todo.service.BusinessException;
import com.renan.todo.service.JwtService;
import com.renan.todo.service.TaskListService;
import com.renan.todo.util.MessageUtil;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

/**
 * The controller for lists of tasks operations
 */
@RequiredArgsConstructor
@RestController
public class TaskListController {

    private final TaskListService taskListService;
    private final JwtService jwtService;

    /**
     * Delete a task list for the user with its tasks
     *
     * @param jwt - token of user
     * @param id  - list id
     */
    @DeleteMapping(value = "/api/taskList/{id}")
    public void deleteTaskList(@RequestHeader(AuthFilter.AUTH) String jwt, @PathVariable("id") Integer id) {
        Integer userID = jwtService.getIdUser(jwt);
        taskListService.deleteTaskList(id, userID);
    }

    /**
     * Update a task list for the user
     *
     * @param jwt  - token of user
     * @param form - input data for list creation
     * @param id   - list id
     */
    @PutMapping(value = "/api/taskList/{id}", consumes = MediaType.APPLICATION_JSON_VALUE)
    public void updateTaskList(@RequestHeader(AuthFilter.AUTH) String jwt, @RequestBody TaskListForm form, @PathVariable("id") Integer id) {
        Integer userID = jwtService.getIdUser(jwt);
        taskListService.updateTaskList(form.getListName(), form.getListDescription(), id, userID);
    }

    /**
     * Create a new task list for the user
     *
     * @param jwt  - token of user
     * @param form - input data for list creation
     *
     * @return - the new list
     */
    @PostMapping(value = "/api/taskList", consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
    public TaskListDTO createTaskList(@RequestHeader(AuthFilter.AUTH) String jwt, @RequestBody TaskListForm form) {
        Integer userID = jwtService.getIdUser(jwt);
        return taskListService.createList(form.getListName(), form.getListDescription(), userID);
    }

    /**
     * get lists and tasks for the user
     *
     * @param jwt - token of user
     *
     * @return - the lists
     */
    @GetMapping(value = "/api/list", produces = MediaType.APPLICATION_JSON_VALUE)
    public List<TaskListDTO> getLists(@RequestHeader(AuthFilter.AUTH) String jwt) {
        Integer userID = jwtService.getIdUser(jwt);
        return taskListService.getLists(userID);
    }

    /**
     * get lists for the user
     *
     * @param jwt - token of user
     *
     * @return - the lists
     */
    @GetMapping(value = "/api/taskList", produces = MediaType.APPLICATION_JSON_VALUE)
    public List<TaskListDTO> getTasksAndLists(@RequestHeader(AuthFilter.AUTH) String jwt) {
        Integer userID = jwtService.getIdUser(jwt);
        return taskListService.getTasksAndLists(userID);
    }

}

File: to-do-app/to-do-kotlin/src/main/kotlin/com/renan/todo/controller/TaskListController.kt
package com.renan.todo.controller

import com.renan.todo.controller.filter.AUTH
import com.renan.todo.controller.form.TaskListForm
import com.renan.todo.dto.ErrorDTO
import com.renan.todo.service.BusinessException
import com.renan.todo.service.JwtService
import com.renan.todo.service.TaskListService
import com.renan.todo.util.getErrorMessage
import org.springframework.http.HttpStatus
import org.springframework.http.MediaType
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestHeader
import org.springframework.web.bind.annotation.RestController

/**
 * Controller for task list requests
 */
@RestController
class TaskListController(val taskListService: TaskListService, val jwtService: JwtService) {

    /**
     * Delete a task list for the user with its tasks
     */
    @DeleteMapping(value = ["/api/taskList/{id}"])
    fun deleteTaskList(@RequestHeader(AUTH) jwt: String?, @PathVariable("id") id: Int?) =
        taskListService.deleteTaskList(id, jwtService.getIdUser(jwt))

    /**
     * Update a task list for the user
     */
    @PutMapping(value = ["/api/taskList/{id}"], consumes = [MediaType.APPLICATION_JSON_VALUE])
    fun updateTaskList(
        @RequestHeader(AUTH) jwt: String?,
        @RequestBody form: TaskListForm,
        @PathVariable("id") id: Int?
    ) = taskListService.updateTaskList(form.listName, form.listDescription, id, jwtService.getIdUser(jwt))

    /**
     * Create a new task list for the user
     */
    @PostMapping(
        value = ["/api/taskList"],
        consumes = [MediaType.APPLICATION_JSON_VALUE],
        produces = [MediaType.APPLICATION_JSON_VALUE]
    )
    fun createTaskList(@RequestHeader(AUTH) jwt: String?, @RequestBody form: TaskListForm) =
        taskListService.createList(form.listName, form.listDescription, jwtService.getIdUser(jwt))


    /**
     * get lists and tasks for the user
     */
    @GetMapping(value = ["/api/list"], produces = [MediaType.APPLICATION_JSON_VALUE])
    fun getLists(@RequestHeader(AUTH) jwt: String?) = taskListService.getLists(jwtService.getIdUser(jwt))

    /**
     * get lists for the user
     */
    @GetMapping(value = ["/api/taskList"], produces = [MediaType.APPLICATION_JSON_VALUE])
    fun getTasksAndLists(@RequestHeader(AUTH) jwt: String?) =
        taskListService.getTasksAndLists(jwtService.getIdUser(jwt))

}